lib/fluent/plugin/in_unix_client.rb

```diff
@@ -160,14 +174,42 @@ def extract_records
 
         pos_read = 0
         while pos_next_delimiter = @buf.index(@delimiter, pos_read)
-          records << @buf[pos_read...pos_next_delimiter]
+          fixed = fix_format(@buf[pos_read...pos_next_delimiter])
+          records << fixed unless fixed.empty?
           pos_read = pos_next_delimiter + @delimiter.size
         end
 
         @buf.slice!(0, pos_read) if pos_read > 0
 
         records
       end
+
+      private
+
+      def fix_format(record)
+        return record if record.empty?
+        return record unless @for_json
+
+        fix_for_json_list_structure(record)
+      end
+
+      def fix_for_json_list_structure(record)
+        if record[0] == "[".freeze && record[-1] == "]".freeze
+          return record
+        end
+
+        if record[0] == "[".freeze
```

Comment:
`.freeze` doesn't seem needed.

Reply:
Thank you for the advice.

I was referring to these examples, but am I using them incorrectly?

https://github.com/fluent/fluentd/blob/b9203165182be15e60574ec6f1ce78df0894b304/lib/fluent/plugin/in_http.rb#L333-L348

Reply:
Fixed.